snooty/main.py

```diff
@@ -153,51 +198,61 @@ def on_update(self, prefix: List[str], page_id: FileId, page: Page) -> None:
             )
 
     def on_update_metadata(
-        self, prefix: List[str], field: Dict[str, SerializableType]
+        self,
+        prefix: List[str],
+        build_identifiers: BuildIdentifierSet,
+        field: Dict[str, SerializableType],
     ) -> None:
         property_name = "/".join(prefix)
+
+        # Construct filter for retrieving build documents
+        document_filter: Dict[str, Union[str, Dict[str, Any]]] = {
+            "page_id": property_name,
+            **construct_build_identifiers_filter(build_identifiers),
+        }
+
         # Write to Atlas if field is not an empty dictionary
         if field:
             self.client[self.db][COLL_METADATA].update_one(
-                {"page_id": property_name}, {"$set": field}, upsert=True
+                document_filter, {"$set": field}, upsert=True
             )
 
-    def on_delete(self, page_id: FileId) -> None:
+    def on_delete(self, page_id: FileId, build_identifiers: BuildIdentifierSet) -> None:
         pass
 
 
-def usage(exit_code: int) -> None:
-    """Exit and print usage information."""
-    print(
-        "Usage: {} <build|watch|language-server> <source-path> <mongodb-url>".format(
-            sys.argv[0]
-        )
-    )
-    sys.exit(exit_code)
+def _generate_build_identifiers(args: Dict[str, Optional[str]]) -> BuildIdentifierSet:
+    identifiers = {}
+
+    identifiers["commit_hash"] = args["--commit"]
+    identifiers["patch_id"] = args["--patch"]
```

Comment:
since we aren't supporting patch ids in the build until Q1, we could remove this code. This will reduce our testing requirements.